### PR TITLE
ci: switch release workflow to github runners

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

Release workflow is still using self-hosted runners.

## What was done?

- Switched runners to ubuntu-24.4
- Use QEMU for arm builds
- Fix Dockerfile for release

## How Has This Been Tested?

Releasing from this branch

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

## Summary by CodeRabbit

- **New Features**
	- Updated Docker image build and publish workflow for improved runner configurations.
	- Added functionality to create and push a manifest list for better image management.
	- Introduced QEMU setup for cross-platform builds on `linux/arm64`.
	- Enhanced build process for Rust dependencies using `cargo chef`.
	- Improved caching mechanisms for dependencies in the build process.
	- Streamlined the build stages for better clarity and efficiency.

### CODE REVIEW

File: .github/workflows/release-docker-image.yml [hunk 1]
```diff
@@ -37,9 +37,9 @@ jobs:
     strategy:
       matrix:
         include:
-          - runner: ["self-hosted", "linux", "x64", "ubuntu-platform"]
+          - runner: ubuntu-24.04
             platform: linux/amd64
-          - runner: ["self-hosted", "linux", "arm64", "ubuntu-platform"]
+          - runner: macos-14
```
[the remark this answers is not in the document]
Comment (shumkov) [reply .github/workflows/release-docker-image.yml]: It is exist

File: Dockerfile [hunk 2]
```diff
@@ -335,7 +332,7 @@ SHELL ["/bin/bash", "-o", "pipefail","-e", "-x", "-c"]
 
 WORKDIR /platform
 
-COPY --from=build-planner /platform/recipe.json /platform/.cargo /platform/
+COPY --from=build-planner --parents /platform/recipe.json /platform/.cargo /
```
[the remark this answers is not in the document]
Comment (shumkov) [reply Dockerfile:335]: All correct

**Review by lklimek — APPROVED** [review]
Looks good, but please check Rabbit comments

File: .github/workflows/release-docker-image.yml [hunk 3]
```diff
@@ -54,6 +54,10 @@ jobs:
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           aws-region: ${{ secrets.AWS_REGION }}
 
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+        if: ${{ matrix.platform == 'linux/arm64' }}
```
[the remark this answers is not in the document]
Comment (shumkov) [reply .github/workflows/release-docker-image.yml:59]: we build devcontainer for both platforms so we need to enable it unconditionally

**Pull request merged**